### ISSUE TITLE
reticulate doesn't respect chunk option fig.process which modifies the path

### ISSUE THREAD

**Issue by Midnighter** [rstudio/reticulate#1007]
The chunk option [`fig.process`](https://yihui.org/knitr/options/#plots) seems to not be respected by reticulate.

I'm using the software carpentries' Rmarkdown template for creating lessons (https://github.com/carpentries/styles). The setup that they have created uses `fig.process` to modify the figure path to use the document's parent directory. Reticulate does not respect the chunk option and so generated figures end up on the wrong path.

I can work around this but it'd be nice if reticulate uses the chunk option properly.

Rmarkdown that demonstrates the issue:

````rmarkdown
---
title: '`fig.process` Demo'
---

```{r setup}
fix_fig_path <- function(pth) file.path("..", pth)
knitr::opts_chunk$set(tidy = FALSE, results = "markup", comment = NA,
               fig.align = "center", fig.path = "fig/rmd-",
               fig.process = fix_fig_path,
               fig.width = 8.5, fig.height = 8.5,
               fig.retina = 2, echo = TRUE)
```

```{python}
import matplotlib.pyplot as plt
import pandas as pd
```

```{python}
df = pd.DataFrame({
  "a": range(10)
})
```

```{python}
df.plot()
```
````

After knitting, the generated document contains an image reference to `fig/rmd-unnamed-chunk-1-1.png` but it should be `../fig/rmd-unnamed-chunk-1-1.png`.

`sessionInfo()`
```
R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Pop!_OS 20.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=de_BE.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=de_BE.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=de_BE.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=de_BE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        lattice_0.20-44   png_0.1-7         digest_0.6.27     rappdirs_0.3.3   
 [6] grid_4.0.5        jsonlite_1.7.2    evaluate_0.14     rlang_0.4.11      Matrix_1.3-3     
[11] reticulate_1.20   rmarkdown_2.8     tools_4.0.5       xfun_0.23         yaml_2.2.1       
[16] compiler_4.0.5    htmltools_0.5.1.1 knitr_1.33
```

**Comment by kevinushey** [comment]
Indeed, only a subset of knitr options are currently supported. To the best of my knowledge, custom engines (like reticulate's) need to implement handling for all potential knitr options by themselves, and this is one of the options we have not yet implemented support for.

This one looks relatively straight-forward so hopefully we can make this happen without too much trouble.

**Comment by Midnighter** [comment]
I had some additional problems with using Altair which would then export to PNG so what I ended up doing was a bit painful:

I chose to duplicate each code chunk that contained a figure. One code chunk is included but not evaluated (`eval=FALSE`) which shows the code "as it should have been". The almost duplicate of that would explicitly set the export path for the PNG and be evaluated but not included in the teaching material (`include=FALSE`). I then directly included figures via markdown. I don't recommend this approach but it works. [Take a look here if you're interested.](https://github.com/data-science-for-biotech/python-pandas-viz-ml/blob/e4d11ae1e7ed5fa3833f78f3049744b722a97464/_episodes_rmd/02b-data-visualization.Rmd#L197-L209)
